src/WalkingModule/src/Module.cpp

```diff
@@ -922,23 +941,80 @@ bool WalkingModule::updateModule()
+            CoMVelocityDesired[1] = m_stableDCMModel->getCoMVelocity().data()[1];
+            CoMVelocityDesired[2] = m_retargetingClient->comHeightVelocity();
+            data.vectors["com::velocity::desired"].assign(CoMVelocityDesired.begin(), CoMVelocityDesired.begin() + CoMVelocityDesired.size());
+
+            // Left foot
+            data.vectors["left_foot::position::measured"].assign(leftFoot.getPosition().data(), leftFoot.getPosition().data() + leftFoot.getPosition().size());
+            data.vectors["left_foot::position::desired"].assign(m_leftTrajectory.front().getPosition().data(), m_leftTrajectory.front().getPosition().data() + m_leftTrajectory.front().getPosition().size());
+            data.vectors["left_foot::orientation::measured"].assign(leftFoot.getRotation().asRPY().data(), leftFoot.getRotation().asRPY().data() + leftFoot.getRotation().asRPY().size());
+            data.vectors["left_foot::orientation::desired"].assign(m_leftTrajectory.front().getRotation().asRPY().data(), m_leftTrajectory.front().getRotation().asRPY().data() + m_leftTrajectory.front().getRotation().asRPY().size());
+
+            // "lf_des_dx", "lf_des_dy", "lf_des_dz",
+            // "lf_des_droll", "lf_des_dpitch", "lf_des_dyaw",
+            data.vectors["left_foot::linear_velocity::measured"].assign(m_leftTwistTrajectory.front().getLinearVec3().data(), m_leftTwistTrajectory.front().getLinearVec3().data() + m_leftTwistTrajectory.front().getLinearVec3().size());
+            data.vectors["left_foot::angular_velocity::measured"].assign(m_leftTwistTrajectory.front().getAngularVec3().data(), m_leftTwistTrajectory.front().getAngularVec3().data() + m_leftTwistTrajectory.front().getAngularVec3().size());
+
+            // "lf_force_x", "lf_force_y", "lf_force_z",
+            // "lf_force_roll", "lf_force_pitch", "lf_force_yaw",
+            data.vectors["left_foot::linear_force::measured"].assign(m_robotControlHelper->getLeftWrench().getLinearVec3().data(), m_robotControlHelper->getLeftWrench().getLinearVec3().data() + m_robotControlHelper->getLeftWrench().getLinearVec3().size());
+            data.vectors["left_foot::angular_torque::measured"].assign(m_robotControlHelper->getLeftWrench().getAngularVec3().data(), m_robotControlHelper->getLeftWrench().getAngularVec3().data() + m_robotControlHelper->getLeftWrench().getAngularVec3().size());
+
+            // Right foot
+            data.vectors["right_foot::position::measured"].assign(rightFoot.getPosition().data(), rightFoot.getPosition().data() + rightFoot.getPosition().size());
+            data.vectors["right_foot::position::desired"].assign(m_rightTrajectory.front().getPosition().data(), m_rightTrajectory.front().getPosition().data() + m_rightTrajectory.front().getPosition().size());
+            data.vectors["right_foot::orientation::measured"].assign(rightFoot.getRotation().asRPY().data(), rightFoot.getRotation().asRPY().data() + rightFoot.getRotation().asRPY().size());
```

Comment:
> The function asRPY() returns a copy of the object. It cannot be used in this way as it produces unexpected results. Indeed, I checked the logged orientation and it contains 7 values instead of 3 as instead expected.

Same here.

Reply:
As above, fixed in https://github.com/robotology/walking-controllers/pull/97/commits/45a8c20cfdb00e85d78d31137deb9d1a87365d65.